webapp/src/components/App/Singlestudy/explore/Modelization/Areas/Hydro/Allocation/Fields.tsx

```diff
@@ -0,0 +1,153 @@
+import { Box } from "@mui/material";
+import { useFieldArray } from "react-hook-form";
+import { useOutletContext } from "react-router";
+import { useCallback, useMemo } from "react";
+import { AxiosError } from "axios";
+import { t } from "i18next";
+import { useSnackbar } from "notistack";
+import Fieldset from "../../../../../../../common/Fieldset";
+import { useFormContextPlus } from "../../../../../../../common/Form";
+import useAppSelector from "../../../../../../../../redux/hooks/useAppSelector";
+import {
+  getAreas,
+  getCurrentAreaId,
+  getStudySynthesis,
+} from "../../../../../../../../redux/selectors";
+import { StudyMetadata } from "../../../../../../../../common/types";
+import { AllocationFormFields, setAllocationFormFields } from "./utils";
+import AllocationSelect from "./AllocationSelect";
+import AllocationField from "./AllocationField";
+import useEnqueueErrorSnackbar from "../../../../../../../../hooks/useEnqueueErrorSnackbar";
+
+export default function Fields() {
+  const {
+    study: { id: studyId },
+  } = useOutletContext<{ study: StudyMetadata }>();
+  const { enqueueSnackbar } = useSnackbar();
+  const enqueueErrorSnackbar = useEnqueueErrorSnackbar();
+  const areaId = useAppSelector(getCurrentAreaId);
+  const areas = useAppSelector((state) => getAreas(state, studyId));
+  const areasById = useAppSelector(
+    (state) => getStudySynthesis(state, studyId)?.areas
+  );
+  const { control, setValue } = useFormContextPlus<AllocationFormFields>();
+  const { fields, append } = useFieldArray({
+    control,
+    name: "allocation",
+  });
+
+  // Get a list of all the area IDs that are allocated
+  const allocatedAreaIds = useMemo(
+    () => fields.map((field) => field.areaId),
+    [fields]
+  );
+
+  // Filter the list of areas to only include those that are not allocated
+  const filteredAreas = useMemo(
+    () => areas.filter((area) => !allocatedAreaIds.includes(area.id)),
+    [areas, allocatedAreaIds]
+  );
+
+  ////////////////////////////////////////////////////////////////
+  // Utils
+  ////////////////////////////////////////////////////////////////
+
+  // Get the name of an area based on its ID.
+  const getAreaLabel = useCallback(
+    (areaId: string) => {
+      return areasById?.[areaId]?.name ?? "";
+    },
+    [areasById]
+  );
+
+  // Append a new allocation field to the form values.
+  const appendField = useCallback(
+    (obj: { areaId: string; coefficient: number }) => {
+      append(obj);
+    },
+    [append]
+  );
+
+  // Remove an allocation field at the specified index from the form values and the server.
+  const removeField = useCallback(
```

Comment:
Not necessary, use `remove` form the API and `onSubmit` will be called. Delete `removeField`.